### DISCUSSION TITLE
How to use external web components?

### DISCUSSION THREAD

**Discussion by kulak** [sycamore-rs/sycamore#587] [Q&A] [answered]
How would one use a custom html element used by web components in builder and macro syntax?

Example:  https://www.webcomponents.org/element/quarkd

```
<!DOCTYPE html>
<html lang="en">
  <body>
    <quark-button loading="false" id="btn">Button</quark-button>
  </body>

  <script src="./node_modules/quarkd/lib/button/index.js" />
  // or cdn
  <script>
    window.addEventListener(function () {
      const el = document.getElementById("btn");
      el.loading = true;
      setTimeout(() => {
        el.loading = true;
      }, 2000);
    });
  </script>
</html>
```

**Comment by kulak** [comment]
I think [documentation](https://sycamore-rs.netlify.app/docs/v0.8/basics/view) has answer for macro based solution, but it is not clear if there is solution for builder.

```
view! { cx,
    // Custom elements!
    my-custom-element {}
}
```

**Comment by sapphire-arches** [comment, marked as answer]
I'm not super familiar with web components, but perhaps `sycamore::builder::tag` is what you're looking for? That will give you a builder that pastes an arbitrary tag in to the generated HTML.
